packages/filesystem/src/browser/filesystem-preferences.ts

```diff
@@ -29,7 +29,7 @@ import { environment } from '@theia/core/shared/@theia/application-package/lib/e
 export const WIN32_MAX_FILE_SIZE_MB = 300; // 300 MB
 export const GENERAL_MAX_FILE_SIZE_MB = 16 * 1024; // 16 GB
 
-export const MAX_FILE_SIZE_MB = environment.electron.is() ? process.arch === 'ia32' ? WIN32_MAX_FILE_SIZE_MB : GENERAL_MAX_FILE_SIZE_MB : 32;
+export const MAX_FILE_SIZE_MB = environment.electron.is() && 'process' in window ? process.arch === 'ia32' ? WIN32_MAX_FILE_SIZE_MB : GENERAL_MAX_FILE_SIZE_MB : 32;
```

Comment:
I think it would be better to have:

```ts
export const MAX_FILE_SIZE_MB = typeof process === 'object'
    ? process.arch === 'ia32'
        ? WIN32_MAX_FILE_SIZE_MB
        : GENERAL_MAX_FILE_SIZE_MB
    : 32;
```